### PR TITLE
BMP180: migrate to new I2C API

### PR DESCRIPTION
**Pull request by orangecms**
I am a bit unsure regarding the "extra" API in the I2C sensors. It looks like it was quite tightly coupled to the old Python desktop app. Let's discuss that in the next meeting.

### CODE REVIEW

File: pslab/external/BMP180.py [hunk 1]
```diff
@@ -1,32 +1,24 @@
-from __future__ import print_function
-
 import time
-
 from numpy import int16
```
Comment (orangecms) [inline pslab/external/BMP180.py:2]: This is quite likely unnecessary, I guess? Some Pythonista help with that would be great. :)

**Comment by orangecms** [comment]
WIth this
```python
from pslab import ScienceLab
from pslab.external.BMP180 import BMP180
from time import sleep

psl = ScienceLab()  # This initializes the I2C bus.

sensor = BMP180()

data = sensor.readTemperature()
print(data)
sleep(2)
data = sensor.readPressure()
print(data)
```

I get
```
calib: 1.2451171875 0.001999969482421875 0.006079673767089844 0.012499809265136719 65535.0 20.400000000000002 1.59375 255 4.98046875 0.194549560546875 65.535 0.0024901963770389557 2.364375 0.9929838180541992 4.42087184637785e-06
12.8
996.8078400249741
```

As of now. `12.8` looks quite odd for the temperature, and I always get that same value across multuple reads, using a BME280. It's definitely not 12.8 Celsius here. Almost [1000 hPa for the pressure is reasonable](https://en.wikipedia.org/wiki/Atmospheric_pressure), but I also get the same value across multiple reads. Something is :fish:y.

File: pslab/external/BMP180.py [hunk 2]
```diff
@@ -1,32 +1,24 @@
-from __future__ import print_function
```
Comment (bessman) [inline pslab/external/BMP180.py:1]: Woops, I accidentally deleted your comment @orangecms! To answer your question, this is indeed not needed. It was for python2 compatibility,

Comment (orangecms) [reply pslab/external/BMP180.py:1]: Right. My comment actually refered to the numpy int16 thing. That looks like a huge dependency for such a little thing, WDYT?

Comment (bessman) [reply pslab/external/BMP180.py:1]: Ah, right. Yes, I can't see any reason why that would be necessary.

(I was trying to delete my own reply to your comment, but somehow managed to delete the whole thread. Sorry about that!)

**Comment by bessman** [comment]
> As of now. `12.8` looks quite odd for the temperature, and I always get that same value across multiple reads, using a BME280. It's definitely not 12.8 Celsius here. Almost [`1000` (hPa) for the pressure is reasonable](https://en.wikipedia.org/wiki/Atmospheric_pressure), but I also get the same value across multiple reads. Something is 🐟y.

From a brief look at the code, it looks like in order to tell the sensor to actually measure anything you need call `initTemperature`/`initPressure`. `readTemperature`/`readPressure` just fetches a register value which is static until a measurement command is sent.

**Comment by orangecms** [comment]
> From a brief look at the code, it looks like in order to tell the sensor to actually measure anything you need call `initTemperature`/`initPressure`. `readTemperature`/`readPressure` just fetches a register value which is static until a measurement command is sent.

Isn't that happening in the init?

> However, both `initTemperature` and `initPressure` are broken because the signature of `I2CSlave.write` changed.

Oh then I did something wrong with that part; what's the correct way to use it now?

**Comment by bessman** [comment]
> Isn't that happening in the init?

It is, but only once, and since both temperature and pressure data is stored in the same register (`__REG_RESULT`) the temperature is immediately overwritten.

> what's the correct way to use it now?

Same as `I2CSlave.read`, i.e. first argument is a list of bytes to write into the register specified by the optional second argument (which defaults to 0x0). So instead of:
```python
def initTemperature(self):
    self.write([self._REG_CONTROL, self._CMD_TEMP])
```
it should be
def initTemperature(self):
    self.write([self._CMD_TEMP], self._REG_CONTROL)
```

**Comment by bessman** [comment]
Data sheet: https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf

**Pull request merged**